sdk/daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/preprocessing/CommandPreprocessor.scala

```diff
@@ -303,20 +289,14 @@ private[lf] final class CommandPreprocessor(
 
   @throws[Error.Preprocessing.Error]
   def unsafePreprocessDisclosedContracts(
-      discs: ImmArray[command.DisclosedContract]
+      discs: ImmArray[FatContractInstance]
   ): ImmArray[speedy.DisclosedContract] = {
     var contractIds: Set[Value.ContractId] = Set.empty
-    var contractKeys: Set[crypto.Hash] = Set.empty
 
     discs.map { disclosedContract =>
       if (contractIds.contains(disclosedContract.contractId))
         throw Error.Preprocessing.DuplicateDisclosedContractId(disclosedContract.contractId)
       contractIds += disclosedContract.contractId
-      disclosedContract.keyHash.foreach { hash =>
```

Comment:
Is think thinking that this cannot happen if the contracts are consistent?

Reply:
As far I can tell this cannot happen anymore. It is a reminder of the original design of explicit disclosure.